src/algebra/gcd_monoid/nat.lean

```diff
@@ -0,0 +1,44 @@
+/-
+Copyright (c) 2021 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import algebra.gcd_monoid.finset
+import number_theory.padics.padic_norm
+
+/-!
+# Basic results about setwise gcds on ℕ
+
+This file proves some basic results about `finset.gcd` on `ℕ`.
+
+## Main results
+* `finset.coprime_of_div_gcd`: The elements of a set divided through by their gcd are coprime.
+
+-/
+
+instance : is_idempotent ℕ gcd_monoid.gcd := ⟨nat.gcd_self⟩
+
+namespace finset
+
+theorem coprime_of_div_gcd (s : finset ℕ) (h : ¬∀ x ∈ s, x = 0) : s.gcd (/ s.gcd id) = 1 :=
```

Comment:
I suspect this would be easier to use as
```suggestion
theorem coprime_of_div_gcd (s : finset ℕ) {x : ℕ} (h : x ∈ s) (hnz : x ≠ 0) : s.gcd (/ s.gcd id) = 1 :=
```